src/lib/storage/index/partial_hash/partial_hash_index_impl.hpp

```diff
@@ -16,21 +17,21 @@ class AbstractSegment;
 class PartialHashIndexTest;
 
 /**
- * Forward iterator that iterates over a tsl::robin_map which maps a DataType to a vector of RowIDs. The iteration
- * process is as if the map would have been flattened and then been iterated.
+ * Forward iterator that iterates over a tsl::sparse_map that maps a DataType to a vector of RowIDs. The
+ * iteration process is as if the map would have been flattened and then been iterated.
  *
  * @tparam DataType The key type of the underlying map.
  */
 template <typename DataType>
-class TableIndexFlattenedSparseMapIterator : public BaseTableIndexIterator {
+class TableIndexSparseMapIterator : public BaseTableIndexIterator {
```

Comment:
What would change if we change the map type to abseil's map? This Iterator name is very specific, but isn't the actual implementation pretty standard?

Reply:
I wonder if `TableIndexIterator` isn't enough. I guess every expects that this iterator might break when a new hash map is used so putting the exact type of hash map has no apparent value if there aren't other iterators we need to differentiate here.
Also, from a user perspective I assume that this iterator is no different when it would be defined on a B-Tree. So I am again wondering if we could get rid of this information that it's an iterator for a hash map.

Reply:
Just stumpled over `* (like and end-iterator)`. I think that's supposed to mean `like an end iterator`. Could have a look please?

Reply:
I renamed the iterators to `TableIndexIterator` and `TableIndexNullValuesIterator`.
How do you like this approach?